Document/CategoryManager.php

```diff
@@ -11,14 +11,127 @@
 
 namespace Sonata\ClassificationBundle\Document;
 
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Sonata\ClassificationBundle\Model\CategoryInterface;
 use Sonata\ClassificationBundle\Model\CategoryManagerInterface;
+use Sonata\ClassificationBundle\Model\ContextInterface;
+use Sonata\ClassificationBundle\Model\ContextManagerInterface;
 use Sonata\CoreBundle\Model\BaseDocumentManager;
 use Sonata\DatagridBundle\Pager\Doctrine\Pager;
 use Sonata\DatagridBundle\ProxyQuery\Doctrine\ProxyQuery;
 
 class CategoryManager extends BaseDocumentManager implements CategoryManagerInterface
 {
     /**
+     * @var CategoryInterface[]
+     */
+    protected $categories;
+
+    /**
+     * @var ContextManagerInterface
+     */
+    protected $contextManager;
```

Comment:
Not your fault but both "context" and "manager" should be avoided in variable names IMO… too vague